--- img.c.orig	2020-12-21 05:20:53.000000000 +0100
+++ img.c	2021-10-01 16:58:23.709569359 +0200
@@ -1023,7 +1023,13 @@
    }
    PUTC('\n', pimg->fh);
 
-   tm = time(NULL);
+   // see https://reproducible-builds.org/docs/source-date-epoch/
+   char *source_date_epoch;
+   /* This assumes that the SOURCE_DATE_EPOCH environment variable will contain
+      a correct, positive integer in the time_t range */
+   if ((source_date_epoch = getenv("SOURCE_DATE_EPOCH")) == NULL ||
+       (tm = (time_t)strtoll(source_date_epoch, NULL, 10)) <= 0)
+         time(&tm);
    if (tm == (time_t)-1) {
       fputsnl(TIMENA, pimg->fh);
    } else if (pimg->version <= 7) {
